### DISCUSSION TITLE
How to assign values?

### DISCUSSION THREAD

**Discussion by PEI0214** [idaholab/moose#28325] [Q&A Modules: Solid mechanics]
Hello, I want to assign a value to a variable in a certain area at a certain time. Is there any way of doing this?

**Comment by GiudGiud** [comment]
You could use a ParsedAux with an expression like:
expression = 'if( abs(t - t_0) < 0.001 && (spatial conditions), the_value_to_assign, regular_value_of_the_variable)'

you can also use the TimesEnableControl for handle the 'certain time" aspect

Can you use a block restriction for the "certain area" ?

**Reply by PEI0214** [reply]
Thank you for your reply, but this is not what I was looking for. What I'm going to do is assign zero values to all the stresses and deformations in a given region. Instead of setting auxiliary variables. I don't know if there's a way to do that.

**Reply by GiudGiud** [reply]
> What I'm going to do is assign zero values to all the stresses

what are the stresses? Arent they material properties?
deformations are displacements right, so they are nonlinear variables?

**Reply by PEI0214** [reply]
Yes, it's a nonlinear variable. I only found a way to set the initial conditions. Is there a way to set it at a certain time?

**Reply by GiudGiud** [reply]
To set  nonlinear variables during the solve, you'll need to create a user object.
There are no auxkernels like for auxiliary variables.

**Reply by PEI0214** [reply]
There's no way to do that right now, is there? If I want to add this feature. Do you have any good suggestions?

**Reply by GiudGiud** [reply]
There are no existing ways. 
You can look at the PointwiseRenormalizeVector userobject. This is probably the best example

**Reply by PEI0214** [reply]
Hello, when I use the restart function. How to combine two output files into one?

**Reply by GiudGiud** [reply]
you can use the solutionUserObject for these mixes
then SOlutionAux can change  the aux variables, solutionIC can be tried for nonlinear variables
